packages/web-api-scan-runner/src/crawl-runner/crawl-runner.spec.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import 'reflect-metadata';
+
+import { Crawler, SimpleCrawlerRunOptions } from 'accessibility-insights-crawler';
+import { GlobalLogger } from 'logger';
+import { Page } from 'puppeteer';
+import { IMock, It, Mock, MockBehavior } from 'typemoq';
+import { ScanMetadataConfig } from '../scan-metadata-config';
+import { ScanMetadata } from '../types/scan-metadata';
+import { CrawlRunner } from './crawl-runner';
+
+// This class exists because trying to mock a Crawler without it
+// caused an unexplained timeout while calling the real Crawler class' constructor
+class CrawlerMock extends Crawler<string[]> {
+    constructor() {
+        super(null);
+    }
+}
+
+type CrawlerProvider = () => Promise<Crawler<string[]>>;
+
+describe('CrawlRunner', () => {
+    const scanId = 'testId';
+    const baseUrl = 'testUrl';
+    const discoveryPatterns = ['testPattern'];
+    const page = {} as Page;
+
+    let loggerMock: IMock<GlobalLogger>;
+    let scanMetaDataConfigMock: IMock<ScanMetadataConfig>;
+
+    beforeEach(() => {
+        loggerMock = Mock.ofType<GlobalLogger>(undefined, MockBehavior.Loose);
+        loggerMock.setup((m) => m.setCommonProperties({ scanId: scanId }));
+
+        scanMetaDataConfigMock = Mock.ofType<ScanMetadataConfig>(undefined, MockBehavior.Strict);
+        scanMetaDataConfigMock.setup((m) => m.getConfig()).returns(() => ({ id: scanId } as ScanMetadata));
+    });
+
+    afterEach(() => {
+        loggerMock.verifyAll();
+        scanMetaDataConfigMock.verifyAll();
+    });
+
+    it('returns undefined when crawler provider returns null', async () => {
+        const crawlerProviderMock = createCrawlerProviderMock(null);
+
+        const crawlRunner = new CrawlRunner(crawlerProviderMock.object, loggerMock.object, scanMetaDataConfigMock.object);
+
+        expect(await crawlRunner.run(baseUrl, discoveryPatterns, page)).toBeUndefined();
+
+        crawlerProviderMock.verifyAll();
```

Comment:
I don't see any verifiable calls set up for this mock, should the call set up in createCrawlerProviderMock be set as verifiable?

Reply:
Mocks are verifiable by default.

Reply:
I mistakenly thought that it wasn't necessary to call verifiable. I conflated that with the fact that it isn't necessary to pass a parameter to verifiable if you intent the function to be called only once.